### DISCUSSION TITLE
Ho do I set/change the boot volume in terraform

### DISCUSSION THREAD

**Discussion by mrndev** [ionos-cloud/terraform-provider-ionoscloud#480] [Q&A] [answered]
Hi,

I have a VM which I want to boot first with a iso image and cdrom (OPNsense). After the first deployment I install opensense on the volume. Everything is fine so far. But, how can I now define the following boots to start from the volume? If I do nothing, the system just always boots from the cdrom. I can of course set the boot flag in the data center designer, but I would like to set the boot partition in the terraform file. Somehow I have not found a way to seti there. Here is my code
```tf
data "ionoscloud_image" "cdrom" {
  name = "OPNsense-23.1-OpenSSL-dvd-amd64-2.iso"
}

resource "ionoscloud_server" "opnsense" {
    count                 = 1
    name                  = "OPNsense ${count.index}"
    datacenter_id         = ionoscloud_datacenter.myvdc.id
    cores                 = 2
    ram                   = 2048
    type                  = "ENTERPRISE"

    # I want the first boot from here
    boot_cdrom = data.ionoscloud_image.cdrom.id

    # after which I want to set my tf file to boot from here. How do I define that?
    volume {
        name              = "OS"
        size              = 5
        disk_type         = "SSD Standard"
    }

    nic {
      lan = ionoscloud_lan.publiclan.id
      ips = [ ionoscloud_ipblock.myblock.ips[count.index] ]
      name = "Public LAN NIC"
    }
}
```

**Comment by cbeti-ionos** [comment, marked as answer]
Hi,

It is not possible currently, to change the boot device from the cdrom to another volume for a server in Terraform.

**Reply by mrndev** [reply]
Hi, thanks for the reply. So I guess the workaround is then to change it in the DCD (manual step) or use ionosctl if it needs to be scripted. Just for the record, for the customers that use iso images (or i guess the problem applies for snapshots too) to boostrap their infrastructure, changing the boot device in terraform would be a nice feature for the future.

**Reply by cbeti-ionos** [reply]
We will investigate this and soon start working on a feature to allow switching between boot devices from terraform.

**Reply by mrndev** [reply]
Great - good to hear that :)
